### PR TITLE
Add checkbox-ce-oem build for series 24 (Infra)

### PR DESCRIPTION
**Pull request by LiaoU3**
## Description
I add classic and uc 24 for checkbox-ce-oem, and I also modify the `latest` channel point to classic 24.

## Resolved issues
N/A

## Documentation
N/A

## Tests
Trial run for the github action
https://github.com/canonical/checkbox/actions/runs/9462277112/job/26064865038

### CODE REVIEW

File: contrib/checkbox-ce-oem/checkbox-ce-oem-snap/series_classic24/snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,103 @@
+name: checkbox-ce-oem
+summary: Checkbox CE OEM test runner and public providers for 24.04 classic
+description: "Checkbox CE OEM test runner and public providers for 24.04 classic"
+confinement: classic
+grade: stable
+
+version: '1.0-jammy'
```
Comment (rickwu666666) [inline contrib/checkbox-ce-oem/checkbox-ce-oem-snap/series_classic24/snap/snapcraft.yaml:7]: This should be Noble
```suggestion
version: '1.0-noble'
```

Comment (rickwu666666) [inline contrib/checkbox-ce-oem/checkbox-ce-oem-snap/series_classic24/snap/snapcraft.yaml:7]: This should be Noble
```suggestion
version: '1.0-noble'
```

File: contrib/checkbox-ce-oem/checkbox-ce-oem-snap/series_uc24/snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,139 @@
+name: checkbox-ce-oem
+summary: Checkbox CE OEM test runner and public providers
+description: "Checkbox CE OEM test runner and public providers"
+confinement: strict
+grade: stable
+
+version: '1.0-jammy'
```
Comment (rickwu666666) [inline contrib/checkbox-ce-oem/checkbox-ce-oem-snap/series_uc24/snap/snapcraft.yaml:7]: This should be Noble
```suggestion
version: '1.0-noble'
```

**Review by rickwu666666 — CHANGES_REQUESTED** [review]
Please see my inline commet.

**Comment by LiaoU3** [comment]
> Please see my inline commet.

@rickwu666666 Thank you for your suggestions. I have fixed my code.

**Review by rickwu666666 — APPROVED** [review]
LGTM

**Review by Hook25 — APPROVED** [review]
I'm not sure this will work actually, you are still using snapcraft 7, can sc7 build core24 snaps?

**Comment by LiaoU3** [comment]
> I'm not sure this will work actually, you are still using snapcraft 7, can sc7 build core24 snaps?

Actually it works. BTW, you could also build series 22 with snapcraft6

**Pull request merged**